pagopa/io-app
# chore: [IOAPPCIT-24] Add a crypto key pair generation test at application startup.

## Short description
This PR adds a crypto key pair generation test at application startup using [io-react-native-crypto](https://github.com/pagopa/io-react-native-crypto) library.

## List of changes proposed in this pull request
- ⚠️ .circleci/config.yml: bumps `node` version from `14.17.4` to `16.19.0`.
- ⚠️ .node-version: bumps `node` version from `14.17.4` to `16.19.0`.
- ⚠️ android/build.gradle: bumps `kotlin` version from `1.5.21` to `1.7.0`.
- yarn.lock: adds `io-react-natiive-crypto` library.
- ios/Podfile.lock: adds `io-react-natiive-crypto` library pod.
- package.json: adds `io-react-natiive-crypto` library.
- ts/sagas/startup.ts: adds the key pair generation.
- ts/sagas/startup/generateCryptoKeyPair.ts: this is the new key generation saga.
- ts/store/reducers/__mock__/backendStatus.ts: update the mock with the new definitions.
- ts/store/reducers/backendStatus.ts: add the selector for the new feature flag to enable/disable the key generation.
- ts/utils/crypto.ts: utility functions for the `io-react-natiive-crypto` library.

## How to test
Run the application on both Android and iOS physical devices against the [io-dev-api-server](https://github.com/pagopa/io-dev-api-server). Try running the application also on simulators: it's not guaranteed that the key generation will be successful, but also a crash-free unsuccessful event is good. 
You must set to `true` this [flag](https://github.com/pagopa/io-dev-api-server/blob/c9485701686abc0f47c4272f05d6fd5246a6cd7e/src/payloads/backend.ts#L78) on [io-dev-api-server](https://github.com/pagopa/io-dev-api-server) to enable the key generation.
Try with different scenarios:
- Try with the flag above in a disabled state with a first onboarding user. No key generation should happen.
- Then enable the flag and restart the application. You should see a key generation attempt.
- Try with the flag above in an enabled state with an already logged-in user: you should see a key generation attempt.
- More up to you ;)
